packages/actions-global/src/actions/handle-action-error.ts

```diff
@@ -1,16 +1,41 @@
+import { Dispatch } from 'redux';
 import {
   STATUS_CODES,
   LOGOUT_REASONS,
+  TStatusCode,
+  TAppNotificationApiError,
 } from '@commercetools-frontend/constants';
 import browserHistory from '@commercetools-frontend/browser-history';
 import showApiErrorNotification from './show-api-error-notification';
 import showUnexpectedErrorNotification from './show-unexpected-error-notification';
 
-export default function handleActionError(error, source) {
-  return dispatch => {
+type ActionError =
+  | Error
+  | {
+      statusCode: TStatusCode;
+      body: {
+        message: string;
+        errors?: TAppNotificationApiError | TAppNotificationApiError[];
+      };
+    };
+
+export default function handleActionError(error: ActionError) {
+  return (
+    dispatch: Dispatch<
+      | ReturnType<typeof showApiErrorNotification>
```

Comment:
This is nice with the `ReturnType`

Reply:
I've seen `ReturnType` being used a lot with `infer`.

Reply:
Yeah, `ReturnType` is built with `infer` 😉

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#type-inference-in-conditional-types